thorlcr/activities/thdiskbase.cpp

```diff
@@ -116,6 +114,36 @@ void CDiskReadMasterBase::serializeSlaveData(MemoryBuffer &dst, unsigned slave)
         CSlavePartMapping::serializeNullMap(dst);
 }
 
+void CDiskReadMasterBase::done()
+{
+    if (!subFileStats.empty())
+    {
+        unsigned numSubFiles = subFileStats.size();
+        for (unsigned i=0; i<numSubFiles; i++)
+        {
+            IDistributedFile *file = queryReadFile(i);
```

Comment:
has this been tested?
I don't think this works, afair, there will be only 1 logical file for the super, not 1 for every sub of the super.

Reply:
I have tested it by reading a superfile with 2 subfiles.  The subfiles numDiskReads attributes for each file was updated- each files numDiskReads value reflected its size.  Is there specific circumstance that this should not work that I should test?

Reply:
ah, I see now, I had forgotten how this worked. addSubFile is called once so 1 entry for the super,
but in CMasterActivity::preStart it expands the readFiles array if it contains super files with the subfiles.